README.md

```diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
 
 ### Installation
 
+This extension includes both a client-side JupyterLab extension and a server-side Jupyter notebook extension. Install these using the command line with
+
 ```bash
+pip install jupyterlab-metadata-service
 jupyter labextension install jupyterlab-metadata-service
 ```
 
+If you are running Notebook 5.2 or earlier, enable the server extension by running
+
+```bash
+jupyter serverextension enable --py --sys-prefix jupyterlab-metadata-service
+```
+
 The latest versions of the following browsers are currently _known to work_:
 
 - Firefox
 - Chrome
 - Safari
 
----
 
 ## Development
 
 ### Contributing
 
+#### Backend Server
+
+TODO: Write instructions for how to contribute to the backend server (e.g. how to run in development mode).
+
+#### Frontend Extension
+
 To contribute to the project, please read the [contributor documentation](CONTRIBUTING.md).
 
 JupyterLab Metadata Service Extensions follows the Jupyter [Community Guides](https://jupyter.readthedocs.io/en/latest/community/content-community.html).
 
-For a development install (requires npm version 4 or later), do the following in the repository directory:
-
+For a development install, do the following in the `frontend/` directory:
 ```bash
-npm install
-npm run build
+# If you don't yet have a conda environemnt for your work on this project, run these two commands:
+conda create -n jlmeta -c conda-forge --override-channels nodejs jupyterlab
+conda activate jlmeta
+
+# Install JS dependencies:
+jlpm install
+
+# Register this extension with JupyterLab in development-mode:
 jupyter labextension link .
 ```
 
-To rebuild the package and the JupyterLab app:
+Now you can run JupyterLab in auto-reload mode, like this:
+```bash
+jupyter lab --watch
+```
 
+If you modify the code in the `frontend/` folder, you'll have to rebuilt it (once rebuilt, JupyterLab will notice and do a partial re-build of itself, from the command above):
 ```bash
-npm run build
-jupyter lab build
+jlpm run build
 ```
 
+If you don't want to run the above command every time you modify code, you can watch the source with the following command:
+```bash
+jlpm run watch
+```
+
+
+
+##### JupyterLab master
+To develop against an unreleased version of `JupyterLab` requires that you add this extension
+as a package inside the `jupyterlab` repo:
+
+```bash
+# clone metadata service
+git clone git@github.com:jupyterlab/jupyterlab-metadata-service.git
+
+# create JupyterLab dev environment
+conda create -n jupyterlab-metadata-service -c conda-forge notebook
+git clone https://github.com/jupyterlab/jupyterlab.git jupyterlab
+cd jupyterlab
+conda activate jupyterlab-metadata-service
+pip install -e .
+
+
+# Add metadata service frontend as sibling package
+jlpm run add:sibling ../jupyterlab-metadata-service/frontend
+cd packages/frontend
+# build metadata frontend
+jlpm run build
+```
+
+Edit the files in the package in `./packages/jupterlab-commenting` and run `jupyter lab --watch --dev` in the 
```

Comment:
not sure if this instructions should be here @saulshanabrook any thoughts?

Reply:
I put the wrong folder here. I fixed it below.